File: 프로그래머스/1/12982. 예산/예산.js
function solution(d, budget) {
    d.sort((a, b) => a - b);
    let totalM = 0;
    for (const [i, money] of d.entries()) {
        if (totalM + money > budget) return i;
        totalM += money;
    }
    return d.length;
}

File: 프로그래머스/1/12982. 예산/예산.java
import java.util.*;
class Solution {
    public int solution(int[] d, int budget) {
        Arrays.sort(d);
        int sum = 0;
        int count = 0;
        for (int money : d) {
            if (sum + money > budget) break;
            sum += money;
            count++;
        }
        return count;
    }
}